store/tikv/region_cache_test.go

```diff
@@ -623,6 +623,96 @@ func (s *testRegionCacheSuite) TestSendFailedInMultipleNode(c *C) {
 	c.Assert(ctxFollower1.Peer.Id, Not(Equals), ctxFollower2.Peer.Id)
 }
 
+func (s *testRegionCacheSuite) TestLabelSelectorTiKVPeer(c *C) {
+	dc1Label := []*metapb.StoreLabel{
+		{
+			Key:   "zone",
+			Value: "dc-1",
+		},
+	}
+	dc2Label := []*metapb.StoreLabel{
+		{
+			Key:   "zone",
+			Value: "dc-2",
+		},
+	}
+	dc3Label := []*metapb.StoreLabel{
+		{
+			Key:   "zone",
+			Value: "dc-3",
+		},
+	}
+	s.cluster.UpdateStoreLabels(s.store1, dc1Label)
+	s.cluster.UpdateStoreLabels(s.store2, dc2Label)
+
+	store3 := s.cluster.AllocID()
+	peer3 := s.cluster.AllocID()
+	s.cluster.AddStore(store3, s.storeAddr(store3))
+	s.cluster.AddPeer(s.region1, store3, peer3)
+	s.cluster.UpdateStoreLabels(store3, dc1Label)
+	// Region have 3 peer, leader located in dc-1, followers located in dc-1, dc-2
+	loc, err := s.cache.LocateKey(s.bo, []byte("a"))
+	c.Assert(err, IsNil)
+	seed := rand.Uint32()
+
+	testcases := []struct {
+		name               string
+		t                  kv.ReplicaReadType
+		labels             []*metapb.StoreLabel
+		expectStoreIDRange map[uint64]struct{}
+	}{
+		{
+			name:   "any Peer,located in dc-1",
+			t:      kv.ReplicaReadMixed,
+			labels: dc1Label,
+			expectStoreIDRange: map[uint64]struct{}{
+				s.store1: {},
+				store3:   {},
+			},
+		},
+		{
+			name:   "any Peer,located in dc-2",
+			t:      kv.ReplicaReadMixed,
+			labels: dc2Label,
+			expectStoreIDRange: map[uint64]struct{}{
+				s.store2: {},
+			},
+		},
+		{
+			name:   "only follower,located in dc-1",
+			t:      kv.ReplicaReadFollower,
+			labels: dc1Label,
+			expectStoreIDRange: map[uint64]struct{}{
+				store3: {},
+			},
+		},
+		{
+			name:   "only leader, shouldn't consider labels",
+			t:      kv.ReplicaReadLeader,
+			labels: dc2Label,
+			expectStoreIDRange: map[uint64]struct{}{
+				s.store1: {},
+			},
+		},
+		{
+			name:   "no label matching, fallback to leader",
+			t:      kv.ReplicaReadMixed,
+			labels: dc3Label,
+			expectStoreIDRange: map[uint64]struct{}{
+				s.store1: {},
+			},
+		},
```

Comment:
Why only leader? I assume it may fallback to any peer.

Reply:
I think this may let the logic more complex. For `GetTiKVRPCContext`, it provides the ability to filter peers by `role` or `labels`, if there is no peer matched, the target peer would be fallback to the leader peer without considering the filter condition.

Reply:
Oh, right.